### PR TITLE
[MASTER] En 1359 widget details master

### PR DESCRIPTION
**Pull request by fedelocci**

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

File: src/state/widgets/selectors.js [hunk 1]
```diff
@@ -24,3 +25,27 @@ export const getTypologyWidgetList = createSelector(getListWidget, widgetList =>
     }
     return acc;
   }, {}));
+export const getWidgetInfo = createSelector([getWidgets], (widget) => {
+  const { info } = widget;
```
Comment (alessioscalici) [inline src/state/widgets/selectors.js]: Why not declaring the `getWidgetsInfo` selector and use that, to be coeherent with all the rest?

File: src/state/widgets/selectors.js [hunk 2]
```diff
@@ -24,3 +25,27 @@ export const getTypologyWidgetList = createSelector(getListWidget, widgetList =>
...
+  if (!isEmpty(info)) {
+    const list = uniq(info.draftUtilizers.map(m => m.pageCode)
+      .concat(info.publishedUtilizers.map(m => m.pageCode)));
+    const data = list.reduce((acc, pageCode) => {
```
Comment (alessioscalici) [inline src/state/widgets/selectors.js]: Why not cleaning up all those property accesses like this:

```javascript
const data = list.reduce((acc, pageCode) => {
      const draft = info.draftUtilizers.find(f => f.pageCode === pageCode);
      const publish = info.publishedUtilizers.find(f => f.pageCode === pageCode);
      acc[pageCode] = {
        pageFullPath: (draft && draft.pageFullPath) || (publish && publish.pageFullPath),
        draft: !isEmpty(draft),
        frameDraft: draft && `${draft.frame} [${draft.frameIndex}]`,
        publish: !isEmpty(publish),
        framePublish: publish && `${publish.frame} [${publish.frameIndex}]`,
      };
      return acc;
    }, {});
```

File: src/ui/widgets/detail/DetailWidgetElement.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+              <Icon
+                type="fa"
+                name={data[info].publish ? 'check' : 'close'}
+              />
+            </td>
+            <td className="DetailWidgetElement__td text-center">{data[info].framePublish}</td>
+            <td className="DetailWidgetElement__td text-center">
+              <Icon
+                type="fa"
+                name={data[info].draft ? 'check' : 'close'}
```
Comment (alessioscalici) [inline src/ui/widgets/detail/DetailWidgetElement.js]: Are you sure data[info] is not undefined? Could you use `get(data, `${info}.draft`)` ?

File: src/ui/widgets/detail/DetailWidgetElement.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { Icon, DropdownKebab } from 'patternfly-react';
+import { LinkMenuItem } from 'frontend-common-components';
+import { ROUTE_PAGE_DETAIL, ROUTE_PAGE_CONFIG } from 'app-init/router';
+
+const DetailWidgetElement = ({ widgetInfo }) => {
+  const renderWidgetInfo = () => {
+    const { data } = widgetInfo;
```
Comment (alessioscalici) [inline src/ui/widgets/detail/DetailWidgetElement.js]: Why not passing data directly as a prop? This way the component has to know about the prop structure...

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Pull request closed without merging**